src/test/java/org/candlepin/controller/PoolManagerFunctionalTest.java

```diff
@@ -98,122 +99,8 @@ public class PoolManagerFunctionalTest extends DatabaseTestFixture {
-            if (pinfo == null || productId == null) {
-                return false;
-            }
-
-            if (productId.equals(pinfo.getId())) {
-                return true;
-            }
-
-            Collection<? extends ProductInfo> providedProducts = pinfo.getProvidedProducts();
-            if (providedProducts != null) {
-                for (ProductInfo ppinfo : providedProducts) {
-                    if (ppinfo != null && productId.equals(ppinfo.getId())) {
-                        return true;
-                    }
-                }
-            }
-
-            // TODO: if we need to factor in derived products, recursively call this function
-            // with the derived product ref.
-
-            return false;
-        }
-
-        @Override
-        public Collection<? extends SubscriptionInfo> getSubscriptionsByProductId(String productId) {
-            return this.submap.values().stream()
-                .filter(sub -> this.productUsesProductId(sub.getProduct(), productId))
-                .collect(Collectors.toList());
-        }
-
-        @Override
-        public boolean hasUnacceptedSubscriptionTerms(String ownerKey) {
-            return false;
-        }
-
-        @Override
-        public void sendActivationEmail(String subscriptionId) {
-            // intentionally left empty
-        }
-
-        @Override
-        public boolean canActivateSubscription(ConsumerInfo consumer) {
-            return true;
-        }
-
-        @Override
-        public void activateSubscription(ConsumerInfo consumer, String email, String emailLocale) {
-            // intentionally left empty
-        }
-
-        @Override
-        public boolean isReadOnly() {
-            return false;
-        }
-    }
-
-    private static class MockProductServiceAdapter implements ProductServiceAdapter {
-        private Map<String, ProductInfo> pmap;
-
-        public MockProductServiceAdapter(Collection<? extends ProductInfo> pinfo) {
-            this.pmap = (pinfo != null ? pinfo.stream() : Stream.<ProductInfo>empty())
-                .filter(Objects::nonNull)
-                .collect(Collectors.toMap(ProductInfo::getId, Function.identity()));
-        }
-
-        public MockProductServiceAdapter(ProductInfo... pinfo) {
-            this(pinfo != null ? List.of(pinfo) : null);
-        }
-
-        @Override
-        public Collection<? extends ProductInfo> getProductsByIds(String ownerKey, Collection<String> ids) {
-            return (ids != null ? ids.stream() : Stream.empty())
-                .map(this.pmap::get)
-                .filter(Objects::nonNull)
-                .collect(Collectors.toList());
-        }
-
-        @Override
-        public CertificateInfo getProductCertificate(String ownerKey, String productId) {
-            return null;
-        }
-    }
-
     private CandlepinPoolManager poolManager;
+    private PoolService poolService;
```

Comment:
These tests should be pulled out into a PoolServiceTest suite instead of keeping them as part of the test suite for the reduced-scope pool manager.

Reply:
Done.